### PR TITLE
Exchange env vars cleanup

### PR DESCRIPTION
**Pull request by kookster**

### CODE REVIEW

File: stacks/apps/exchange.yml [hunk 1]
```diff
@@ -517,24 +522,6 @@ Resources:
               ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/fixer-secret
             - Name: GOOGLE_MAPS_API_KEY
               ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/google-maps-api-key
-            - Fn::If:
-                - IsProduction
-                - Name: MAILCHIMP_API_KEY
```
Comment (kookster) [inline stacks/apps/exchange.yml:522]: We don't use mailchimp anymore

File: stacks/apps/exchange.yml [hunk 2]
```diff
@@ -517,24 +522,6 @@ Resources:
...
-                  ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/mailchimp-api-key
-                - !Ref AWS::NoValue
-            - Fn::If:
-                - IsProduction
-                - Name: MAILCHIMP_LIST_ID
-                  ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/mailchimp-list-id
-                - !Ref AWS::NoValue
-            - Name: PAYPAL_BUSINESS_ADDRESS
```
Comment (kookster) [inline stacks/apps/exchange.yml:530]: And we don't use PayPal (all stripe, all the time)

File: stacks/apps/exchange.yml [hunk 3]
```diff
@@ -555,22 +542,12 @@ Resources:
               ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/recaptcha-secret-key
             - Name: RECAPTCHA_SITE_KEY
               ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/recaptcha-site-key
-            - Fn::If:
-                - IsProduction
-                - Name: REDIS_URL
```
Comment (kookster) [inline stacks/apps/exchange.yml:560]: This was for PMP integration, which is gone.

File: stacks/apps/exchange.yml [hunk 4]
```diff
@@ -555,22 +542,12 @@ Resources:
...
-                - !Ref AWS::NoValue
             - Name: SOLR_URL
               ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/solr-url
             - Name: STRIPE_API_KEY
               ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/stripe-api-key
             - Name: STRIPE_PUBLISHABLE_KEY
               ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/stripe-publishable-key
-            - Fn::If:
-                - IsProduction
-                - Name: USE_REPLICA
```
Comment (kookster) [inline stacks/apps/exchange.yml:571]: We used to have writes to one DB. and reads from another instance - no more!

File: stacks/apps/exchange.yml [hunk 5]
```diff
@@ -720,12 +705,8 @@ Resources:
               Value: "80"
             - Name: RAILS_ENV
               Value: !Ref EnvironmentTypeLowercase
-            - Name: NEW_HOMEPAGE
```
Comment (kookster) [inline stacks/apps/exchange.yml:723]: Years ago, we had an updated homepage, no longer need this feature flag.

File: stacks/apps/exchange.yml [hunk 6]
```diff
@@ -467,12 +476,8 @@ Resources:
...
```
Comment (kookster) [inline stacks/apps/exchange.yml:470]: Old feature flag, unused now

File: stacks/apps/exchange.yml [hunk 7]
```diff
@@ -467,12 +476,8 @@ Resources:
...
-              Value: "true"
             - Name: CACHE_STORE
               Value: mem_cache_store
-            - Name: PAYPAL_ACTION_URL
```
Comment (kookster) [inline stacks/apps/exchange.yml:474]: no more PayPal, all stripe

File: stacks/apps/exchange.yml [hunk 8]
```diff
@@ -517,24 +522,6 @@ Resources:
...
-                - Name: MAILCHIMP_LIST_ID
-                  ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/mailchimp-list-id
-                - !Ref AWS::NoValue
-            - Name: PAYPAL_BUSINESS_ADDRESS
-              ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/paypal-business-address
-            - Name: PAYPAL_CERTIFICATE_ID
-              ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/paypal-certificate-id
-            - Name: PAYPAL_NOTIFY_HOST
-              ValueFrom: !Sub /prx/${EnvironmentTypeAbbreviation}/Spire/Exchange/paypal-notify-host
-            - Name: PMP_OP_ID
```
Comment (kookster) [inline stacks/apps/exchange.yml:536]: No more PMP

File: stacks/apps/exchange.yml [hunk 9]
```diff
@@ -1537,7 +1464,7 @@ Resources:
     Properties:
       DelaySeconds: 0
       MessageRetentionPeriod: 345600 # 4 days
-      QueueName: !Sub ${Constants.SqsQueuePrefix}_exchange_porter_callback
+      QueueName: !Sub ${Constants.SqsQueuePrefix}porter_callback
```
Comment (kookster) [inline stacks/apps/exchange.yml:1467]: this resulted in `dc51b3fd_prod_exchange__exchange_porter_callback`, which is not awesome, duplicate `_exchange`

Comment (kookster) [reply stacks/apps/exchange.yml:1467]: This does go with a change to the exchange app's config/messaging.rb as well.

File: stacks/apps/exchange.yml [hunk 10]
```diff
@@ -1532,6 +1459,22 @@ Resources:
         - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
         - { Key: prx:dev:application, Value: Exchange }
       VisibilityTimeout: 5400 # 1.5 hour
+  ExchangePorterCallbackQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      DelaySeconds: 0
+      MessageRetentionPeriod: 345600 # 4 days
+      QueueName: !Sub ${Constants.SqsQueuePrefix}porter_callback
```
Comment (kookster) [inline stacks/apps/exchange.yml:1467]: The old queue is incorrectly named: `dc51b3fd_prod_exchange__exchange_porter_callback`
This creates a new correctly named: `dc51b3fd_prod_exchange_porter_callback`

**Review by farski — APPROVED** [review]
You can get rid of the quotes around `"production.prxtransfer.org"` and `"testing.prxtransfer.org"` inside the `If`s if you want, but not necessary

**Review by farski — APPROVED** [review]

**Pull request merged**